stl/inc/xmemory

```diff
@@ -756,8 +758,9 @@ struct _Default_allocator_traits { // traits for std::allocator
 };
 
 _EXPORT_STD template <class _Alloc>
-struct allocator_traits : conditional_t<_Is_default_allocator<_Alloc>::value, _Default_allocator_traits<_Alloc>,
-                              _Normal_allocator_traits<_Alloc>> {};
+struct _NO_SPECIALIZATIONS allocator_traits : conditional_t<_Is_default_allocator<_Alloc>::value,

```

Comment:
Pedantically speaking, program-defined specialization of `allocator_traits` is only forbidden since C++23. But it's possibly a good thing to "extend" the restriction to old modes.